src/v/datalake/datalake_manager.cc

```diff
@@ -370,4 +371,40 @@ void datalake_manager::stop_translator(const model::ntp& ntp) {
     });
 }
 
+ss::future<uint64_t> datalake_manager::disk_usage() {
+    const auto path = config::node().datalake_staging_path();
+
+    if (!co_await ss::file_exists(path.string())) {
+        co_return 0;
+    }
+
+    chunked_vector<std::filesystem::path> files;
+    co_await directory_walker::walk(
+      path.string(), [&files, path](const ss::directory_entry& de) {
+          if (de.type == ss::directory_entry_type::regular) {
+              files.push_back(path / std::filesystem::path(de.name));
+          }
+          return ss::now();
+      });
```

Comment:
curious is there no utility to compute the size of a directory? Do we always have to walk the directory  and compute sizes of individual files?

Reply:
file systems generally don't provide roll-up directory statistics (and in particular, I don't believe XFS does). so the only solutions are application-specific. what you can do here (and what we do for storage layer) is attempt to cache the size, which works well for segments that get created and then exist unchanged for long periods of time. for this scratch space files, there is probably lots of churn so it's not clear that the caching approach will have any benefit.